torch_geometric/utils/augmentation.py

```diff
@@ -0,0 +1,204 @@
+                [3., 4., 5.],
+                [0., 1., 2.]])
+        >>> node_perm
+        tensor([2, 1, 0])
+    """
+    if not training:
+        perm = torch.arange(x.size(0), device=x.device)
+        return x, perm
+    perm = torch.randperm(x.size(0), device=x.device)
+    return x[perm], perm
+
+
+def mask_feature(x: Tensor, p: float = 0.5, mode: str = 'col',
+                 training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Randomly masks feature from the feature matrix
+    :obj:`x` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`x`, (2) the feature
+    mask broadcastable with :obj:`x` (mode='row' and mode='col') or
+    with the same shape as :obj:`x` (mode='all'), indicating where features
+    are retained.
+
+    Args:
+        x (FloatTensor): The feature matrix.
+        p (float, optional): The masking ratio. (default: :obj:`0.5`)
+        mode (str, optional): The masked scheme to use for feature masking.
+            (:obj:`"row"`, :obj:`"col"` or :obj:`"all"`). (default: :obj:`col`)
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    :rtype: (:class:`FloatTensor`, :class:`BoolTensor`)
+
+    Examples:
+
+        >>> # Masked features are column-wise sampled
+        >>> x = torch.tensor([[1, 2, 3],
+        ...                   [4, 5, 6],
+        ...                   [7, 8, 9]], dtype=torch.float)
+        >>> x, feat_mask = mask_feature(x)
+        >>> x
+        tensor([[1., 0., 3.],
+                [4., 0., 6.],
+                [7., 0., 9.]]),
+        >>> feat_mask
+        tensor([[True, False, True]])
+
+        >>> # Masked features are row-wise sampled
+        >>> x, feat_mask = mask_feature(x, mode='row')
+        >>> x
+        tensor([[1., 2., 3.],
+                [0., 0., 0.],
+                [7., 8., 9.]]),
+        >>> feat_mask
+        tensor([[True], [False], [True]])
+
+        >>> # Masked features are uniformly sampled
+        >>> x, feat_mask = mask_feature(x, mode='all')
+        >>> x
+        tensor([[0., 0., 0.],
+                [4., 0., 6.],
+                [0., 0., 9.]])
+        >>> feat_mask
+        tensor([[False, False, False],
+                [True, False,  True],
+                [False, False,  True]])
+    """
+    if p < 0. or p > 1.:
+        raise ValueError(f'Masking ratio has to be between 0 and 1 '
+                         f'(got {p}')
+    if not training or p == 0.0:
+        return x, torch.ones_like(x, dtype=torch.bool)
+    assert mode in ['row', 'col', 'all']
+
+    if mode == 'row':
+        mask = torch.rand(x.size(0), device=x.device) >= p
+        mask = mask.view(-1, 1)
+    elif mode == 'col':
+        mask = torch.rand(x.size(1), device=x.device) >= p
+        mask = mask.view(1, -1)
+    else:
+        mask = torch.randn_like(x) >= p
+
+    x = x.masked_fill(~mask, 0.)
```

Comment:
Let's support `fill_value` in args. Sometimes we may want to fill it with different values such as mean.

Reply:
```suggestion
    x = x.masked_fill(~mask, fill_value)
```

Reply:
That is interesting but probably not possible to compute for `mode="all"`, right? In addition, do we need to support re-scaling of features as in the `dropout` case?

Reply:
> That is interesting but probably not possible to compute for mode="all"

We can let `fill_value` be a user-specified value with 0 as default?

> do we need to support re-scaling of features as in the dropout case?

That's not common in literature and I don't think we need to do that, the masked features are simply treated as missing ones.

Reply:
`fill_value` with 0 as default sounds good to me.

Re-scaling also seems great. We can support re-scaling. Maybe set it as `False` by default.

IMO, filling 0 does not work well when the data is not normalized. So some way to keep the mean unchanged could be helpful.